apisix/admin/ssl.lua

```diff
@@ -138,6 +156,22 @@ function _M.post(id, conf)
         return 400, err
     end
 
+    -- encrypt private key
+    local local_conf = core.config.local_conf()
+    local iv
+    if local_conf and local_conf.apisix
+       and local_conf.apisix.ssl.key_encrypt_salt then
+        iv = local_conf.apisix.ssl.key_encrypt_salt
+    end
+
+    local aes_128_cbc_with_iv = (type(iv)=="string" and #iv == 16) and
+            assert(aes:new(iv, nil, aes.cipher(128, "cbc"), {iv=iv})) or nil
+
+    if aes_128_cbc_with_iv ~= nil then
+        local encrypted = aes_128_cbc_with_iv:encrypt(conf.key)
+        conf.key = ngx_encode_base64(encrypted)
+    end
+
```

Comment:
the same codes as https://github.com/apache/incubator-apisix/pull/1678/files#diff-ef2c372bb70b36a8baed8b3ed9818244R106-R120, we should write in a function